kubectl-plugin/pkg/cmd/submit/submit.go

```diff
@@ -0,0 +1,486 @@
+package submit
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/kubectl/pkg/cmd/portforward"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/google/shlex"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/spf13/cobra"
+
+	rayv1api "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+)
+
+type SubmitJobOptions struct {
+	ioStreams        *genericiooptions.IOStreams
+	configFlags      *genericclioptions.ConfigFlags
+	RayJob           *unstructured.Unstructured
+	submissionID     string
+	entryPoint       string
+	cluster          string
+	fileName         string
+	workingDir       string
+	runtimeEnv       string
+	address          string
+	entryPointCPU    float32
+	entryPointGPU    float32
+	entryPointMemory int
+	noWait           bool
+}
+
+type RayJob struct {
+	*rayv1api.RayJob
+}
+
+func NewSubmitJobOptions(streams genericiooptions.IOStreams) *SubmitJobOptions {
+	return &SubmitJobOptions{
+		ioStreams:   &streams,
+		configFlags: genericclioptions.NewConfigFlags(true),
+	}
+}
+
+func NewSubmitJobCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewSubmitJobOptions(streams)
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:   "submit [OPTIONS] -- ENTRYPOINT",
+		Short: "Submit ray job to ray cluster",
```

Comment:
It's better to also set `Long` and `Example`, especially for `Example`, because this command is a little bit complicated.

Reply:
I added `Long` and `Example` please take a look.